packages/libs/deer/src/number.rs

```diff
@@ -57,6 +57,17 @@ impl Number {
     const fn float(n: f64) -> Self {
         Self(OpaqueNumber::Float(n))
     }
+
+    /// # Safety
+    ///
+    /// The caller must verify that the value is a string of a valid integer or floating point
+    /// number, otherwise the [`Number`] returned from this function will lead to undefined
+    /// behaviour.
```

Comment:
Why would this be UB? Isn't the worst thing, that could happen, an `.unwrap()` on a `None` value? If yes, can we remove the `unsafe` and say that in the docs? In any case, is there a debug assertion we can put here? (I guess not)

Reply:
Not necessarily in the number code, which will always return `None` instead. The reason is that other implementations may choose to use the `Display` output for the number value. In that case (because we do not check if the value is a number) they would potentially fail, even though they expected a valid `Number`. I do not know if that qualifies as `unsafe`, as the values contained in a `Number` might not be a number.

Reply:
Hmm, okay, let's leave it in, then. This also enables us to do more magic here if desired at some point. As we still have to explicitly allow `unsafe` and no `unsafe` code is used, it's fine to leave `miri` disabled. However, do we want enable `miri` for subcrates? (For simplicity we could just enable it for `deer`)

Reply:
enabled in [c440c96](https://github.com/hashintel/hash/pull/1478/commits/c440c96e7710ddc39d378c5a44b21bd320eaaf20)